src/tests/AsyncTests.fs

```diff
@@ -200,18 +200,22 @@ open Fable.Core
 
 [<Test>]
 let ``Interaction between Async and Promise works``() =
-    let res = ref false
-    async { res := true }
-    #if FABLE_COMPILER
-    |> Async.StartAsPromise
-    |> Async.AwaitPromise
-    #else
-    |> Async.StartAsTask
-    |> Async.AwaitTask
-    #endif
-    |> Async.StartImmediate
-    equal true !res
-    
+    async {
+        let res = ref false
+        async { res := true }
+        #if FABLE_COMPILER
+        |> Async.StartAsPromise
+        |> Async.AwaitPromise
+        #else
+        |> Async.StartAsTask
+        |> Async.AwaitTask
+        #endif
+        |> Async.StartImmediate
+        // some wait is needed on Linux for some reason
```

Comment:
Technically I should have wrapped the Async.Sleep like that, cause it's only happening in clr, not js:

#if !FABLE_COMPILER
do! Async.Sleep 50
#endif

For some reason the issue is not happening under Windows.
There were some recent changes in Async.AwaitTask but I don't think that's it.